### PR TITLE
Add autoreload functionality to jupyter notebooks

### PR DESCRIPTION
**Pull request by KrasserKai**
This fixes issue #189 I proposed. With these changes, each Jupyter Notebook uses the "autoreload" functionality to reload modules when code is executed. This makes it more convenient when developing a module while simultaneously testing the functionality in the Jupyter Notebook, since you don't need to restart the kernel each time a change is made to the module.

### CODE REVIEW

File: notebooks/naive_bayes.ipynb [hunk 1]
```diff
@@ -774,7 +784,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.10.11"
+   "version": "3.12.3"
```
Comment (pkeilbach) [inline notebooks/naive_bayes.ipynb:787]: This is probably related to your local setup. Please remove this from the PR ✌🏼 (similar for the other places below)

**Review by pkeilbach — CHANGES_REQUESTED** [review]
This is a really cool enhancement, @KrasserKai 🚀 

This is worth 2 bonus points!

Before we merge, let's make sure that the PR only contains the actual change (see comment) ✌🏼

File: notebooks/naive_bayes.ipynb (hunk 1 above)
Comment (KrasserKai) [reply notebooks/naive_bayes.ipynb]: Ah yes the problem was that I had executed some cells in those notebooks and so the stored metadata of the output changed. I have restored the files to their old output, so now the only change is the added code cell with the autoreload.

**Review by pkeilbach — APPROVED** [review]
LGTM! 🏅 🏅

**Pull request merged**